### ISSUE TITLE
gtk and gst not working after update

### ISSUE THREAD

**Issue by CCTV-1** [Alexpux/MINGW-packages#4466]
today update gtk(3.24.0-1 -> 3.24.1-1),gst(1.14.2-1 -> 1.14.3-1) and related dependence

may tools don't work,console message is as follows:

GStreamer-WARNING: 
```
Failed to load plugin 'D:\MSYS2\mingw64\lib\gstreamer-1.0\libgstdtls.dll'
Failed to load plugin 'D:\MSYS2\mingw64\lib\gstreamer-1.0\libgstshout2.dll'
Failed to load plugin 'D:\MSYS2\mingw64\lib\gstreamer-1.0\libgstsrtp.dll'
```
Gtk-WARNING:
```
drawing failure for widget 'GtkDrawingArea': out of memory
drawing failure for widget 'GtkGrid': out of memory
drawing failure for widget 'GtkWindow': out of memory
```

**Comment by lazka** [comment]
The gstreamer warnings are expected due to #4458

I can't reproduce the gtk errors, everything looks fine here.

**Comment by CCTV-1** [comment]
the code like the follows:
```
#include <cstdlib>
#include <cstdint>

#include <vector>
#include <functional>

#include <gtk/gtk.h>

typedef std::vector<std::pair<std::function<std::string(void)>, std::function<void(void)>>> Menu_t;

static void draw_menu(cairo_t *cairo, Menu_t &items, GtkWidget *widget);
gboolean draw_window(GtkWidget *widget, cairo_t *cario, gpointer data);

GdkPixbuf *pixbuf = NULL;

int main(int argc, char *argv[])
{
    Menu_t items;
    items.push_back({[]() { return std::string("test string"); },
                     gtk_main_quit});
    gtk_init(&argc, &argv);

    GtkWidget *window = gtk_window_new(GTK_WINDOW_TOPLEVEL);

    GtkWidget *area = gtk_drawing_area_new();

    pixbuf = gdk_pixbuf_new_from_file("test.jpg", NULL);
    int window_height = gdk_pixbuf_get_height(pixbuf);
    int window_width = gdk_pixbuf_get_width(pixbuf);
    gtk_window_set_default_size(GTK_WINDOW(window), window_height, window_width);
    gtk_window_set_resizable(GTK_WINDOW(window), FALSE);

    g_signal_connect(G_OBJECT(area), "draw", G_CALLBACK(draw_window), &items);

    g_signal_connect(G_OBJECT(window), "destroy", G_CALLBACK(gtk_main_quit), NULL);
    gtk_container_add(GTK_CONTAINER(window), area);
    gtk_widget_show_all(window);
    gtk_main();

    g_object_unref(pixbuf);

    return EXIT_SUCCESS;
}

gboolean draw_window(GtkWidget *widget, cairo_t *cairo, gpointer data)
{
    Menu_t &items = *((Menu_t *)data);

    gdk_cairo_set_source_pixbuf(cairo, pixbuf, 0, 0);
    cairo_paint(cairo);

    draw_menu(cairo, items, widget);
    return TRUE;
}

static void draw_menu(cairo_t *cairo, Menu_t &items, GtkWidget *widget)
{
    int widget_height = gdk_pixbuf_get_height(pixbuf);
    int widget_width = gdk_pixbuf_get_width(pixbuf);
    const int box_start_x = widget_height / 6;
    const int box_start_y = widget_width / 6;

    double box_height = widget_height * 2 / 3;
    double box_width = widget_width * 2 / 3;
    cairo_move_to(cairo, box_start_x, box_start_y);
    cairo_set_source_rgba(cairo, 43.0 / 255, 42.0 / 255, 43.0 / 255, 0.7);
    cairo_rel_line_to(cairo, 0, box_height);
    cairo_rel_line_to(cairo, box_width, 0);
    cairo_rel_line_to(cairo, 0, 0 - box_height);
    cairo_rel_line_to(cairo, 0 - box_width, 0);
    cairo_close_path(cairo);
    cairo_set_line_width(cairo, 2);
    cairo_fill_preserve(cairo);
    cairo_set_source_rgba(cairo, 0, 0, 0, 1.0);
    cairo_stroke(cairo);

    size_t item_total = items.size();
    double item_size = box_height / item_total;
    PangoFontDescription *font_desc = pango_font_description_from_string("Microsoft YaHei UI 18");
    cairo_pattern_t *pattern = cairo_pattern_create_linear(0.0, 0.0, widget_width, widget_height);
    for (size_t i = 0; i < item_total; i++)
    {
        std::string name = items[i].first();
        PangoLayout *layout = gtk_widget_create_pango_layout(widget, name.c_str());
        pango_layout_set_font_description(layout, font_desc);
        int layout_width = 0;
        int layout_height = 0;
        pango_layout_get_pixel_size(layout, &layout_width, &layout_height);
        int content_start_x = (box_width - layout_width) / 2 + box_start_x;
        int content_start_y = box_start_y + i * item_size + (item_size - layout_height) / 2;
        cairo_move_to(cairo, content_start_x, content_start_y);
        pango_cairo_layout_path(cairo, layout);
        cairo_pattern_add_color_stop_rgb(pattern, 0.0, 1.0, 1.0, 1.0);
        cairo_set_source(cairo, pattern);
        cairo_fill_preserve(cairo);
        cairo_set_line_width(cairo, 0.5);
        cairo_stroke(cairo);
        g_object_unref(layout);
    }
    cairo_pattern_destroy(pattern);
    pango_font_description_free(font_desc);
}

```

the strange thing is if `"test string"` change to `"teststring"`,the work.

**Comment by lazka** [comment]
I can reproduce the warning, but also with older gtk versions (3.24.0, 3.22.30). Please report this upstream: https://gitlab.gnome.org/GNOME/gtk/issues

**Comment by lazka** [comment]
thanks, https://gitlab.gnome.org/GNOME/gtk/issues/1370

The gstreamer errors should be fixed

**Comment by CCTV-1** [comment]
Mmmh,now gstreamer there is still one error.
```
$ gst-play-1.0 D:/test.mp3

(gst-play-1.0:880): GStreamer-WARNING **: 08:02:56.777: Failed to load plugin 'D:\MSYS2\mingw64\lib\gstreamer-1.0\libgstshout2.dll': 'D:\MSYS2\mingw64\lib\gstreamer-1.0\libgstshout2.dll'
```

**Comment by lb90** [comment]
A fix for the errors in the gtk side has landed in cairo 15.6, released yesterday.
see: commit [3a03c1ba](https://gitlab.freedesktop.org/cairo/cairo/commit/3a03c1ba4bc30492721abdb6c9ec8868663ef237)

**Comment by lazka** [comment]
Thanks. cairo 1.16 is in the repo now.
